app/models/miq_report.rb

```diff
@@ -144,6 +144,24 @@ def self.display_filter_details(cols, mode)
     end
   end
 
+  def list_schedules
+    exp = MiqExpression.new("=" => {"field" => "MiqReport.id",
+                                    "value" => id})
+    MiqSchedule.filter_matches_with exp
+  end
+
+  def add_schedule(data)
+    data['name'] ||= name
```

Comment:
Could we `dup` `data` here or otherwise create a new hash so that `data` is not mutated in this method?